website/docs/cloud-docs/api-docs/oauth-clients.mdx

```diff
@@ -412,10 +456,104 @@ curl \
   https://app.terraform.io/api/v2/oauth-clients/oc-XKFwG6ggfA9n7t1K
 ```
 
+## Attach an OAuth Client to projects
+
+`POST /oauth-clients/:id/relationships/projects`
+
+| Parameter | Description                                                                                        |
+| --------- | -------------------------------------------------------------------------------------------------- |
+| `:id`     | The ID of the oauth client to attach to projects. Use the "List OAuth Clients" endpoint to find IDs. |
+
+| Status  | Response                  | Reason                                                                     |
+| ------- | ------------------------- | -------------------------------------------------------------------------- |
+| [204][] | Nothing                   | The request was successful                                                 |
+| [404][] | [JSON API error object][] | OAuth Client not found or user unauthorized to perform action                |
+| [422][] | [JSON API error object][] | Malformed request body (one or more projects not found, wrong types, etc.) |
+
+### Request Body
+
+This POST endpoint requires a JSON object with the following properties as a request payload.
+
+Properties without a default value are required.
+
+| Key path | Type           | Default | Description                                                                                                                                                                                                                                                          |
+| -------- | -------------- | ------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `data[]` | array\[object] |         | A list of resource identifier objects that defines which projects to attach the oauth client to. These objects must contain `id` and `type` properties, and the `type` property must be `projects` (e.g. `{ "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }`).    |
```

Comment:
```suggestion
| `data[]` | array\[object] |         | A list of resource identifier objects that defines which projects are associated with the OAuth client. If `data.attributes.organization-scoped` is set to `false`, the OAuth client is only available to this list of projects. Each object in this list must contain a project `id` and use the `"projects"` type. For example, `{ "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }`.    |
```

Reply:
Also there's no default value given here, while above `[]` is given as the default value. Should it be `[]` here as well?

Reply:
I have added default value for both attach and detach.